### DISCUSSION TITLE
plugin-proposal-decorators

### DISCUSSION THREAD

**Discussion by fed-gren** [babel/babel#14149] [Q&A] [answered]
I'm using @ babel / plugin-proposal-class-properties and @ babel / plugin-proposal-decorators together.

```json
"plugins": [
      ["@babel/proposal-class-properties", {"loose": true}],
      ["@babel/plugin-proposal-decorators", { "legacy": true }],
      ...
]
```

There are orders to apply the plugin to the [official documentation](https://babeljs.io/docs/en/babel-plugin-proposal-decorators#note-compatibility-with-babelplugin-proposal-class-properties).

I'm curious about the reason. Why apply @ babel / plugin-proposal-decorators first?

**Comment by JLHwung** [comment, marked as answer]
> Why apply @ babel / plugin-proposal-decorators first?

Decorators proposal is based on properties proposal. So it has to be transpiled before properties. In other words, Babel can not transpile class properties but preserving decorators. And browsers support class properties earlier than decorators, so you can transpile decorators but preserving properties if your target already supports them.

> There are orders to apply the plugin to the [official documentation](https://babeljs.io/docs/en/babel-plugin-proposal-decorators#note-compatibility-with-babelplugin-proposal-class-properties).

BTW the example you posted is the wrong order example.

**Reply by fed-gren** [reply]
Thanks!!
